Nidhi_Gupta/progress.md

```diff
@@ -104,35 +104,85 @@ Retrospect on Week 7:
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-403317501)
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-404215861)
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-405013606)
+
+Retrospect on Week 9:
+This week I worked upon making comments on responses PR backward compatible. I enjoyed discussing with mentors about ways we can ensure that legacy data is handled. I saw few of the old data migration scripts which was pretty amazing. I previously thought that data migration is something extremely complicated. But now I wish to write one myself for comments feature in future. In my GQR PR, there were logic holes in my previous implementation which got discovered during review. So I think I need to be more careful and think deeply so as to avoid writing bad logic in future.  I couldn't complete this week's plan because there is still lot of room for improvement in comments PR and current week's plan can only be completed after main PR gets merged.
+
 Week 10 Plan:
 - [ ] Implement the ability for respondents to add comments for rubric questions
 - [ ] Implement the ability for respondents to add comments for distribute questions
 - [ ] Update UI for response comments
 - [ ] Add tests for both
 
+Works
+- Merged PR [Student: view results: rank questions: show average #8701](https://github.com/TEAMMATES/teammates/pull/8784)
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-405386280)
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-405716741)
+- Review ready PR [Student: submit responses: allow student to add a comment for each response #7239](https://github.com/TEAMMATES/teammates/pull/8830#issuecomment-406586982)
+
+Retrospect on Week 10:
+This week I worked upon separating feedback participant comments from instructor comments completely. It simplified the implementation of comment indexes. I also wrote lots of unit tests for all the new methods that I introduced in my multiple PRs. I feel more confident about writing tests at the end of this week. No major difficulties encountered. The reason for not completing this week's plan is same as the week before. Unfortunately I forgot to update my project plan last week.
+
 Week 11 Plan:
-- [ ] Finish any remaining work from week 8-10
-- [ ] Fix issues after code review
+- [x] Finish any remaining work from week 8-10
+- [x] Fix issues after code review
 - [ ] Implement backend of ability for instructors to disable comments
```

Comment:
I updated project plan of this and following weeks before they started. (You can verify from my commits in this PR) I hope that's okay.